docs/docsite/rst/presentations/openstack-modules-intro.rst

```diff
@@ -0,0 +1,394 @@
+
+This talk is Free Software
+==========================
+
+* Written for presentty (https://pypi.python.org/pypi/presentty)
+* docs/docsite/rst/presentations/openstack-modules-intro.rst
+
+Ansible OpenStack Modules
+=========================
+
+* http://docs.ansible.com/ansible/list_of_cloud_modules.html#openstack
+* focused on consuming OpenStack APIs
+* Some are end-user focused
+* Some are deployer focused
+
+Work on All OpenStack Clouds
+============================
+
+Don't let the existence of Rackspace modules fool you. The OpenStack
+modules work just great on Rackspace.
+
+Work Around Deployer Differences ... To a Point
+===============================================
+
+* Herculean effort is made to hide a giant amount of differences
+* Even so - some bugs are EWONTFIX
+
+A provider decided to redefine the OpenStack AZ concept complete
+with API changes. That is unsupportable.
+
+Based on Shade
+==============
+
+* abstracts deployment differences
+* designed for multi-cloud
+* simple to use
+* massive scale
+
+  * optional advanced features to handle 20k servers a day
+
+* Initial logic/design extracted from nodepool
+* Librified to re-use in Ansible
+
+Integration Testing
+===================
+
+* Every shade patch is integration tested in OpenStack Infra
+* Every shade patch tests the ansible modules in Infra too
+* Support coming soon to test PRs to modules in Infra
+
+Let's Take a Few Steps Back
+===========================
+
+OpenStack in Ansible and Multi-cloud Operations are easy...
+
+but you need to know a few things.
+
+* Module structure
+* Terminology
+* Config
+
+Module Structure
+================
+
+* All modules start with `os_`
+* End-user oriented modules are named for the resource
+
+  * os_image
+  * os_sever
+
+* Operator oriented modules are named for the service
+
+  * os_nova_flavor
+  * os_keystone_domain
+
+* If it could go both ways, we prefer user orientation
+
+  * os_user
+  * os_group
+
+* If more than one service provides a resource, they DTRT
+
+  * os_security_group
+  * os_floating_ip
+
+OpenStack Dynamic Inventory Script
+==================================
+
+* github.com/ansible/ansible/contrib/inventory/openstack.py
+* Uses all hosts in all clouds as one inventory
+* Excludes hosts without IPs
+* Makes a bunch of auto-groups (flavor, image, az, region, cloud)
+* Makes groups for `metadata['groups']`
+
+Modules for All OpenStack Resources are Welcome Upstream
+========================================================
+
+* If there is a thing you want to manage without a module, add one
+* We're very welcoming/accepting
+* Same goes for features to existing modules
+
+To Serve All Users, We Do Have to be Strict
+===========================================
+
+* Modules that touch OpenStack APIs must do so via `shade`
+* Other modules don't need to (ex: `os_tempest_config`)
+* In user modules, vendor differences should be hidden
```

Comment:
"must do so" what? must serve all users? The next bullet doesn't make sense as a standalone--other modules don't need to do what?